system/jlib/jlog.cpp

```diff
@@ -2324,6 +2324,41 @@ MODULE_EXIT()
     thePassAllFilter = NULL;
 }
 
+#ifdef _CONTAINERIZED
+#define LOGFIELDS_ATT "@fields"
+#define LOGMSGDETAIL_ATT "@detail"
+#define LOGMSGAUDIENCES_ATT "@audiences"
+#define LOGMSGCLASSES_ATT "@classes"
+
+void setupContainerizedLogMsgHandlerFromConfig()
+{
+    IPropertyTree * logconfig = queryComponentConfig().queryPropTree("logging");
```

Comment:
if "logging" isn't defined, what is the default behaviour?
It still uses the stderr handler and all fields, and what trace level?

Can we define defaults here in the absence of settings, e.g.:
```
    // defaults
    unsigned msgFields = MSGFIELD_time|MSGFIELD_date|MSGFIELD_milliTime|MSGFIELD_msgID|MSGFIELD_process|MSGFIELD_thread|MSGFIELD_code|MSGFIELD_quote|MSGFIELD_prefix|MSGFIELD_audience;
    unsigned msgAudiences = MSGAUD_all;
    unsigned msgClasses = MSGAUD_all;
    LogMsgDetail logDetail = DefaultDetail;
```

then if (logging) and any of the above present, override these and then run the setMessageFields/changeMonitorFilter lines unconditionally.

Reply:
We could define defaults already, but they're already defined in void setDefaults()
The logfields are only overwritten if the configuration contains @fields, no need to default.

If either  the audiences, classes, or details are specified, the other fields are defaulted to the same value as setDefaults()